File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        unordered_map<int, int> count;

        for(int num: nums) {
            count[num]++;
        }

        vector<vector<int>> bucket(nums.size() + 1);

        for(auto &it:count) {
            bucket[it.second].push_back(it.first);
        }

        vector<int> result;

        for(int i=nums.size(); i>=0 && result.size() < k;i--) {
            for(int num: bucket[i]) {
                result.push_back(num);
                if(result.size() == k) {
                    return result;
                }
            }
        }

        return result;
    }
};

File: Data Structures & Algorithms/climbing-stairs/submission-0.py
class Solution:
    def climbStairs(self, n: int) -> int:
        if n == 1:
            return 1
        elif n == 2:
            return 2
        
        first = 1
        second = 2

        for i in range(3, n+1):
            third = second + first
            first = second
            second = third
        
        return second